packages/grid/src/vaadin-grid-mixin.js

```diff
@@ -1039,6 +1045,43 @@ export const GridMixin = (superClass) =>
       }
     }
 
+    /** @private */
+    __isTargetInViewport(target) {
+      const targetRect = target.getBoundingClientRect();
+      const gridRect = this.getBoundingClientRect();
+
+      if (targetRect.left < gridRect.left || targetRect.right > gridRect.right) {
+        // Target cell is not fully in the viewport.
+        return false;
+      }
+
+      const frozen = [...target.parentNode.children].find((cell) => cell.hasAttribute('last-frozen'));
+      if (frozen && frozen !== target) {
```

Comment:
if target is a frozen cell but it's not the last frozen one, this would result with `false`. We should probably always return true if the target cell is frozen.